source/codegen/metadata_mutation.py

```diff
@@ -50,3 +50,51 @@ def mark_non_proto_params(parameters):
 def get_size_param(param, parameters):
     named_params = { p['name'] : p for p in parameters }
     return named_params.get(param['size']['value'], None)
+
+def add_attribute_values_enums(enums, attribute_enums, service_class_prefix):
+    """Update enums metadata to add new enums that will be used by SetAttribute* APIs."""
+    mapping_types = {
+        'ViInt64': 'std::int64_t',
+        'ViReal64': 'double',
+        'ViString': 'std::string'
+    }
+    for type_name in attribute_enums:
+        values_to_create = {}
+        for enum_name in sorted(attribute_enums[type_name]):
+            enum = enums[enum_name]
+            for value in enum["values"]:
+                value_name = value['name'].replace(f"{service_class_prefix.upper()}_", f"{common_helpers.pascal_to_snake(enum_name).upper()}_")
+                value_value = value["value"]
+                values_to_create[value_name] = value_value
+        type_enum_name = get_attribute_values_enum_name(service_class_prefix, type_name)
+        enum_value_prefix = (f"{service_class_prefix}_{type_name[2:]}").upper()
+        generate_mappings = type_name in mapping_types
```

Comment:
`type_name in mapping_types and values_to_create`

Reply:
Even if there are no `values_to_create`, we do need to create mappings dictionary. Otherwise we will have weird typecasting on server. for e.g. if there are no values to create for ViStringAttributeValues enum, typecasting on server would try to typecast `request->attribute_value()`, which is an enum to ViString and throw compile time error.

Reply:
Updated to early return if there are no `values_to_create`.